### ISSUE TITLE
Couldn't compile hpack on Ubuntu 14.04 (elixir)

### ISSUE THREAD

**Issue by LanfordCai** [joedevivo/chatterbox#79]
I created a new project by `mix new hpack_test`, and added only one dependence: `{:chatterbox, git: "git://github.com/joedevivo/chatterbox.git"}`, and then `mix do deps.get, compile`, an error raised:

```
Compiled src/lager_format.erl
==> hpack (compile)
src/hpack.erl:none: error in parse transform 'lager_transform': {function_clause,
                                             [{lager_transform,
                                               '-walk_ast/2-fun-0-',
                                               [{typed_record_field,
                                                 {record_field,19,
                                                  {atom,19,
                                                   connection_max_table_size},
                                                  {integer,19,4096}},
                                                 {type,19,non_neg_integer,
                                                  []}}],
                                               [{file,
                                                 "src/lager_transform.erl"},
                                                {line,60}]},
                                              {lists,map,2,
                                               [{file,"lists.erl"},
                                                {line,1239}]},
                                              {lists,map,2,
                                               [{file,"lists.erl"},
                                                {line,1239}]},
                                              {lager_transform,walk_ast,2,
                                               [{file,
                                                 "src/lager_transform.erl"},
                                                {line,60}]},
                                              {compile,
                                               '-foldl_transform/2-anonymous-2-',
                                               2,
                                               [{file,"compile.erl"},
                                                {line,958}]},
                                              {compile,foldl_transform,2,
                                               [{file,"compile.erl"},
                                                {line,960}]},
                                              {compile,
                                               '-internal_comp/4-anonymous-1-',
                                               2,
                                               [{file,"compile.erl"},
                                                {line,315}]},
                                              {compile,fold_comp,3,
                                               [{file,"compile.erl"},
                                                {line,341}]}]}
Compiling src/hpack.erl failed:
ERROR: compile failed while processing /root/Projects/hpack_test/deps/hpack: rebar_abort
** (Mix) Could not compile dependency :hpack, "/root/.mix/rebar compile skip_deps=true deps_dir="/root/Projects/hpack_test/_build/dev/lib"" command failed. You can recompile this dependency with "mix deps.compile hpack", update it with "mix deps.update hpack" or clean it with "mix deps.clean hpack"
```

Could you give me some suggestions?

**Comment by hpopp** [comment]
The chatterbox `lager` dependency is locked at 2.2.1, but it's had a [recent fix to support Erlang/OTP 19](https://github.com/basho/lager/pull/321). @joedevivo

**Comment by LanfordCai** [comment]
@hpopp Oh, thank you!

And I think I should add some notes here:
My working environment is Ubuntu 16.04, elixir 1.3.1

It is a bit wierd that when I added chatterbox into dependences by `{:chatterbox, git: "git://github.com/joedevivo/chatterbox.git"}`, older version would be fetch (which one have lager 2.2.1), but if I use `{:chatterbox, github: "joedevivo/chatterbox"}`, a right version would be fetch. But an error would raised: `In deps/hpack/rebar.config: ... {:lager, ~r/2.2.1 ...}`, however, I found that hpack have also dumped to lager 2.2.3, quite strange... So at last I have to override hpack in my `mix.exs`:

```elixir
{:chatterbox, github: "joedevivo/chatterbox"},
{:hpack, github: "joedevivo/hpack", override: true},
```

And everything works fine.
